spec/machine/RISCV64/MachineOps.thy

```diff
@@ -491,4 +491,68 @@ definition setVSpaceRoot :: "paddr \<Rightarrow> machine_word \<Rightarrow> unit
   "setVSpaceRoot pt asid \<equiv> machine_op_lift $ setVSpaceRoot_impl pt asid"
 
 end
+
+lemma mult_le_mono':
+  "\<lbrakk>a \<le> a'; b \<le> b'; a' * b' \<le> c\<rbrakk> \<Longrightarrow> a * b \<le> (c :: nat)"
+  using le_trans mult_le_mono by blast
+
+notepad begin
```

Comment:
This is nice. For extra peace of mind, it'd be better to put this before the `axiomatization` and `consts'` statements that introduce `factor1` etc to make it really obvious that these aren't accidentally used in here (I don't think they can, but the more obvious the better).

Should probably also add a comment that this notepad shows that the axioms introduced later are consistent.

Reply:
Sure, I'll add a comment and put this before the `axiomatization`. 

Are you happy for it to be put in the arch dependent files? I can say in the comment that the proofs are currently independent of the architecture

Reply:
I think it's fine to duplicate this one. Usually we'd want to avoid duplication, but there is a lot of it in MachineOps already and now trying hard for this specific bit doesn't really add much more.

We probably do want a deeper reform of these definitions at some point, but not now.